src/app/qgsvectorlayerproperties.cpp

```diff
@@ -249,6 +249,12 @@ QgsVectorLayerProperties::QgsVectorLayerProperties(
   diagLayout->addWidget( diagramPropertiesDialog );
   mDiagramFrame->setLayout( diagLayout );
 
+  // WMS Name as layer short name
+  mLayerShortNameLineEdit->setText( layer->shortName() );
+  // WMS Name validator
+  QRegExp shortNameRx( "^[A-Za-z][A-Za-z0-9\._-]*" );
```

Comment:
The `QRegExp`, just to avoid duplication since the same RE is used in more than one place in the same class. It might be a static member of `QgsVectorLayerProperties`